components/Fields/Number.js

```diff
@@ -1,12 +1,15 @@
-import React, { PureComponent } from 'react';
+import { withProps, compose } from 'recompose';
 import TextInput from './Text';
 
-class NumberInput extends PureComponent {
-  render() {
-    return (
-      <TextInput type="number" {...this.props} />
-    );
-  }
-}
+const withNumericChangeHandler = withProps(props => ({
+  type: 'number',
+  input: {
+    ...props.input,
+    onChange: e => props.input.onChange(parseInt(e.target.value, 10)),
```

Comment:
We might want to check for `NaN` in these; otherwise React issues a warning and the field value is cleared entirely. (Example: accidentally type dash instead of 0: "3-".)

Reply:
I pushed the change suggested above; if you disagree (or don't like the new behavior in NC), please feel free to revert.